lib/api/include/irods/ticket_administration.hpp

```diff
@@ -0,0 +1,598 @@
+#ifndef IRODS_TICKET_ADMINISTRATION_HPP
+#define IRODS_TICKET_ADMINISTRATION_HPP
+
+#undef RxComm
+
+#ifdef IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+#    define RxComm RsComm
+
+struct RsComm;
+#else
+#    define RxComm RcComm
+
+struct RcComm;
+#endif // IRODS_TICKET_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include <string_view>
+#include <type_traits>
+#include <string>
+
+///
+/// \brief Namespace to hold the ticket administration features
+///
+namespace irods::experimental::administration::ticket
+{
+    ///
+    /// \brief Struct tag to indicate admin privilige
+    ///
+    inline struct admin_tag
+    {
+    } admin;
+
+    ///
+    /// \brief Enumeration to indicate the ticket type
+    ///
+    enum class type
+    {
+        read,
+        write
+    };
+
+    ///
+    /// \brief Enumeration to indicate the property of ticket
+    ///
+    enum class property
+    {
+        uses_count,
+        write_file,
+        write_byte
+    };
+
+    ///
+    /// \brief Struct that holds the user name of the user that should be added/removed
+    ///
+    struct users_constraint
+    {
+        const std::string_view name;
+    };
+
+    ///
+    /// \brief  Struct that holds the name of the group that should be added/removed
+    ///
+    struct groups_constraint
+    {
+        const std::string_view name;
+    };
+
+    ///
+    /// \brief  Struct that holds the name of the host that should be added/removed
+    ///
+    struct hosts_constraint
+    {
+        const std::string_view name;
+    };
+
+    ///
+    /// \brief Struct to hold info about the ticket property desired for the ticket specified in further function calls
+    ///
+    struct ticket_property_constraint
+    {
+        const property desired_property;
+        int value_of_property = -1;
+    };
+
+    ///
+    /// \brief A function to run the ticket operation
+    ///
+    /// \param[in] conn The communication Object
```

Comment:
Minor: You could argue that this is an in/out param because it is a non-const reference.

Reply:
In some of the previous examples on iRODS, the communication object seems to be only a [in].